spec/take_away_ui_spec.rb

```diff
@@ -0,0 +1,232 @@
+require 'take_away_ui'
+
+describe TakeAway do
+
+  #Doubling cuisine class
+  let(:dummy_cuisine_klass) {double :dummy_cuisine_klass}
+  subject(:menu) {described_class.new dummy_cuisine_klass, dummy_order_klass, dummy_basket_klass}
+  let(:italian_menu) {double :italian_menu}
+  let(:dinner) {double :dinner}
+  let(:main) {double :main}
+  let(:default_menu_dinner_mains) {{ Mains: 'Price', Margherita: 13, Pasta: 12, Risotto: 11, Calzone: 10, Carbonara: 12, Bolognese: 12, Gnocchi: 11 }}
+
+  #Doubling order class
+  let(:dummy_order_klass) {double :dummy_order_klass}
+  let(:dummy_order) {double :dummy_order}
+
+  #Doubling basket class
+  let(:dish) {double :dish}
+  let(:quantity) {double :quantity}
+  let(:dummy_basket_klass) {double :dummy_basket_klass}
+  let(:dummy_basket) {double :dummy_basket}
+
+  #Doubling checkout class
+  let(:checkout_message) {double :checkout_message}
+  # let(:checkout_time) {Checkout::HOURS_UNTIL_DELIVERY}
+
+
+before do
+  allow(dummy_cuisine_klass).to receive(:new).and_return(italian_menu)
+  allow(italian_menu).to receive(:dishes).and_return(default_menu_dinner_mains)
+end
+
+  describe '#initialize' do
+
+    it 'has the default cuisine choice of ItalianFood' do
+      expect(menu.cuisine_choice).to eq dummy_cuisine_klass
+    end
+
+    it 'has an uninstantiated Order object' do
+      expect(menu.order).to eq dummy_order_klass
+    end
+
+    it 'has an uninstantiated Basket object' do
+      expect(menu.basket).to eq dummy_basket_klass
+    end
+
+    it 'has a selection array' do
+      expect(menu.selection).to be_a Array
+    end
+
+    it 'has an empty selection array as default' do
+      expect(menu.selection).to be_empty
+    end
+
+  end
+
+  context "Choosing and displaying menus" do
+
+    describe '#menu_choice' do
+
+      it 'will raise error "We only serve lunch and dinner" if another meal is asked for' do
+        expect{menu.menu_choice(:breakfast)}.to raise_error("We only serve lunch and dinner")
+      end
+
+      it 'will raise error "Please choose starter, main or dessert" if another meal is asked for' do
+        expect{menu.menu_choice(:dinner, :snack)}.to raise_error("Please choose starter, main, dessert or 'all'")
```

Comment:
Metrics/LineLength: Line is too long. [113/80] (https://github.com/bbatsov/ruby-style-guide#80-character-limits)